projects/components/src/table/table.component.ts

```diff
@@ -278,6 +279,15 @@ export class TableComponent
       this.columnConfigsSubject.next(this.buildColumnConfigs());
     }
 
+    if (changes.selections && this.selections) {
+      // unmark the selections which don't exist in existing row selections
+      this.rowSelections.forEach(row => {
+        this.selections!.includes(row) ? row.$$state.selected = true : row.$$state.selected = false
```

Comment:
Please flip the ternary - no expressions as assignments. But aside from that, don't we need to set $$state.selected for all rows? this would unset it in an existing selection, but would it set it in a new selection?